src/app.rs

```diff
@@ -47,6 +47,138 @@ pub struct App {
     display_help: bool,
     pub is_searching: bool,
     pub channel_text_width: usize,
+    receipt_handler: ReceiptHandler,
+}
+
+#[derive(Debug, Default, PartialEq, Eq)]
+pub struct ReceiptHandler {
+    receipt_set: HashMap<Uuid, ReceiptQueues>,
+    time_since_update: u64,
+}
+
+impl ReceiptHandler {
+    pub fn new() -> Self {
+        Self {
+            receipt_set: HashMap::new(),
+            time_since_update: 0u64,
+        }
+    }
+
+    pub fn add_receipt_event(&mut self, event: ReceiptEvent) {
+        // Add a new set in the case no receipt had been handled for this contact
+        // over the current session
+        self.receipt_set
+            .entry(event.uuid)
+            .or_insert_with(ReceiptQueues::new);
```

Comment:
This method returns `&mut Value`, so you could store in a variable and do the modification from line 74. This would make the second lookup obsolte.

Reply:
Yes ofc, totally missed that one!